app/src/main/java/fi/sabriina/urbanhuikka/MainActivity.kt

```diff
@@ -161,7 +161,9 @@ class MainActivity : AppCompatActivity(), OnCardSwipeListener {
             val selectedCard = gameStateViewModel.getSelectedCard()
             if ( selectedCard != null) {
                 currentCard = selectedCard
-                cardView.setCard(selectedCard)
+                val assistingPlayer = gameStateViewModel.drawAssistingPlayer()
```

Comment:
Assisting player is now selected for all cards instead of just the ones including multiple players. This could be optimized by setting an identifier to cards that require multiple players.